construction.tex

```diff
@@ -1,17 +1,12 @@
 \section{Construction}\label{section:construction}
 
-\import{./}{algorithms/alg.bitcoin.tex}
-
-\todo{explain the bitcoin protocol here ~gtklocker}
+We outline the blockchain address protocol for Bitcoin pay to public key hash (P2PKH)~\cite{bitcoin-dev-guide} transactions. The implementation is parameterized by a secure signature scheme $S$, and a hash function $H$.
+$\GenAddress$ uses $S$ to generate a keypair, and hashes the public key in order to generate the public key hash. A tuple consisting of the public key hash and the secret key is returned.
+$\SpendVerify$ takes a spending transaction $m$, a signature $\sigma$ and a public key hash $pkh$. The signature should contain the public key $pk$ corresponding to $pkh$ such that $H(pk) = pkh$ and a valid signature for the outpoint $m$, $\sigma'$. If these conditions are met, the function returns true, otherwise it returns false.
+The implementation is illustrated in Algorithm~\ref{alg.bitcoin}.
```

Comment:
I wouldn't call this an "implementation". Maybe "The blockchain address protocol is illustrated..."?